db/comdb2.c

```diff
@@ -2413,22 +2413,25 @@ struct dbenv *newdbenv(char *dbname, char *lrlname)
     if (!dbenv->basedir) {
         logmsg(LOGMSG_FATAL, "DB directory is not set in lrl\n");
         return NULL;
-    } else if (gbl_create_mode) {
-        /* make sure the database directory exists! */
+    } 
+
+    if (gbl_create_mode) {
+        logmsg(LOGMSG_DEBUG, "gbl_create_mode is on, "
+               "creating the database directory exists %s\n", dbenv->basedir);
         rc = mkdir(dbenv->basedir, 0774);
         if (rc && errno != EEXIST) {
             logmsg(LOGMSG_ERROR, "mkdir(%s): %s\n", dbenv->basedir,
                    strerror(errno));
-            /* continue, this will make us fail later */
-        }
-    } else {
-        struct stat sb;
-        stat(dbenv->basedir, &sb);
-        if (!S_ISDIR(sb.st_mode)) {
-            logmsg(LOGMSG_FATAL, "DB directory '%s' does not exist\n",
-                   dbenv->basedir);
-            return NULL;
         }
+    } 
+    
+    /* make sure the database directory exists! */
+    struct stat sb;
+    stat(dbenv->basedir, &sb);
```

Comment:
I think I saw this failed before in my branch. I think maybe we need to check rc here for stat. On errors, sb doesnt seem to be initialized.